lib/cloud_controller/deployment_updater/calculators/find_interim_web_process.rb

```diff
@@ -0,0 +1,37 @@
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Calculators
+      class FindInterimWebProcess
+        attr_reader :deployment, :app
+
+        def initialize(deployment)
+          @deployment = deployment
+          @app = deployment.app
+        end
+
+        def call
+          # Find newest interim web process that (a) belongs to a SUPERSEDED (DEPLOYED) deployment and (b) has at least
+          # one running instance.
+          app.web_processes_dataset.
+            qualify.
+            join(:deployments, deploying_web_process_guid: :guid).
+            where(deployments__state: DeploymentModel::DEPLOYED_STATE).
+            where(deployments__status_reason: DeploymentModel::SUPERSEDED_STATUS_REASON).
+            order(Sequel.desc(:created_at), Sequel.desc(:id)).
```

Comment:
**question** It was like this previously, but do you know why the order is important if we're just cleaning these processes up?

Reply:
This is used in the Cancel action, so the goal here is to find a previous web process to restore state to. So once an interim process is found, then that process is scaled UP to the desired number of instances immediately.

Reply:
That said if it does revert to the most recent Deployment, I don't see anywhere that updates the worker/non-web processes to have the same droplet 🤔